checkout/src/main/java/com/checkout/tokenization/usecase/RiskInstanceProvider.kt

```diff
@@ -0,0 +1,38 @@
+package com.checkout.tokenization.usecase
+
+import android.content.Context
+import com.checkout.base.model.Environment
+import com.checkout.risk.Risk
+import com.checkout.risk.RiskConfig
+import com.checkout.risk.RiskEnvironment
+
+internal object RiskInstanceProvider {
+    private var riskInstance: Risk? = null
+
+    suspend fun provide(
+        context: Context,
+        publicKey: String,
+        environment: Environment,
+    ): Risk? {
+        if (riskInstance != null) {
+            return riskInstance
+        }
+
+        val riskEnvironment =
+            when (environment) {
+                Environment.PRODUCTION -> RiskEnvironment.PRODUCTION
+                Environment.SANDBOX -> RiskEnvironment.SANDBOX
+            }
+
+        riskInstance =
+            Risk.getInstance(
+                context,
+                RiskConfig(
+                    publicKey = publicKey,
+                    environment = riskEnvironment,
+                    framesMode = true,
+                ),
+            )
+        return riskInstance
```

Comment:
nit: can you please write some unit test for this part?

Reply:
It might not be very feasible to mock some implementations of `context` and/or the Risk instance creation for the Provider, even with using `mockk()`. We also have a test covering what is directly used (the RiskUsecase). I am happy to take a deeper look/conversation with you afterwards if you have suggestions or workarounds, but can we add this in later as it might not be a blocker for the first release to keep things moving, especially because of the importance of the project and timing?